runtime/src/accounts.rs

```diff
@@ -416,6 +438,36 @@ impl Accounts {
         }
     }
 
+    fn get_requested_accounts_data_size_limit(tx: &SanitizedTransaction) -> Result<u64> {
```

Comment:
I think it would be more appropriate to add a new `ComputeBudget` instruction to let transactions request a new accounts data size limit. I think it's useful for validators to be able to statically analyze how much compute or memory each transaction can use separately. When a validator is replaying a block, they can parallelize lots of transactions that use compute but if they are memory constrained they may not wish to parallelize transactions that load lots of account data.

Reply:
Mmm, it does sound a cleaner option, can add in separate PR as pre-requisite. 

Do we still want to tie data size to compute units somehow? My gut is compute unit is for CPU, it doesn't necessary need to capture memory usage. If a new instruction is added to request data size, it doesn't have to do with compute. But it can/should still be part of fee calculation.

Reply:
#28084 adds `SetAccountsDataSizeLimit` instruction